sof-defconfig

```diff
@@ -23,26 +23,3 @@ CONFIG_SND_SOC_SOF_CANNONLAKE_SUPPORT=y
 CONFIG_SND_SOC_SOF_COFFEELAKE_SUPPORT=y
 CONFIG_SND_SOC_SOF_COMETLAKE_SUPPORT=y
 CONFIG_SND_SOC_SOF_ICELAKE_SUPPORT=y
-CONFIG_SND_SOC_SOF_HDA_LINK=y
-
-# Intel machine drivers
-CONFIG_SND_SOC_INTEL_HASWELL_MACH=m
-CONFIG_SND_SOC_INTEL_BDW_RT5677_MACH=m
-CONFIG_SND_SOC_INTEL_BROADWELL_MACH=m
-CONFIG_SND_SOC_INTEL_BYTCR_RT5640_MACH=m
-CONFIG_SND_SOC_INTEL_BYTCR_RT5651_MACH=m
-CONFIG_SND_SOC_INTEL_CHT_BSW_RT5672_MACH=m
-CONFIG_SND_SOC_INTEL_CHT_BSW_RT5645_MACH=m
-CONFIG_SND_SOC_INTEL_CHT_BSW_MAX98090_TI_MACH=m
-CONFIG_SND_SOC_INTEL_CHT_BSW_NAU8824_MACH=m
-CONFIG_SND_SOC_INTEL_BYT_CHT_DA7213_MACH=m
-CONFIG_SND_SOC_INTEL_BYT_CHT_ES8316_MACH=m
-CONFIG_SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH=m
-CONFIG_SND_SOC_INTEL_BXT_RT298_MACH=m
-CONFIG_SND_SOC_INTEL_BXT_PCM512x_MACH=m
-CONFIG_SND_SOC_INTEL_BXT_WM8804_MACH=m
-CONFIG_SND_SOC_INTEL_GLK_RT5682_MAX98357A_MACH=m
-CONFIG_SND_SOC_INTEL_SOF_RT5682_MACH=m
-
-# Disable HDAudio legacy
-# CONFIG_SND_HDA_INTEL is not set
```

Comment:
Should we keep # CONFIG_SND_HDA_INTEL is not set here? Otherwise, we will run legacy HDA driver in ``merge_config.sh .config base-defconfig sof-defconfig nocodec-defconfig`` cmd.